velox/common/base/BitUtil.h

```diff
@@ -691,8 +691,19 @@ bool inline hasIntersection(
       });
 }
 
-inline int32_t countLeadingZeros(uint64_t word) {
-  return __builtin_clzll(word);
+template <typename T = uint64_t>
+inline int32_t countLeadingZeros(T word) {
+  static_assert(std::is_same_v<T, uint64_t> || std::is_same_v<T, __uint128_t>);
+  if (word == 0) {
```

Comment:
nit: perhaps, add a comment to explain that __builtin_clzll doesn't allow zero input. This would help future readers who may wonder why this explicit check is needed.